drivers/net/ppp.c

```diff
@@ -576,6 +576,20 @@ static int ppp_send(const struct device *dev, struct net_pkt *pkt)
 			protocol = htons(PPP_IP);
 		} else if (net_pkt_family(pkt) == AF_INET6) {
 			protocol = htons(PPP_IPV6);
+#if defined(CONFIG_PPP_DIALUP)
+		} else if (net_pkt_family(pkt) == AF_PACKET) {
```

Comment:
Instead of this, please do
`} else if (IS_ENABLED(CONFIG_PPP_DIALUP) && net_pkt_family(pkt) == AF_PACKET) {`
instead so that the code in the block gets compiler checked even if it is not compiled in.